moon/cfc_chat_transit/server/init.moon

```diff
@@ -2,33 +2,42 @@ require "gwsockets"
 require "cfclogger"
 
 import lshift from bit
+import Read from file
 import GetColor from team
 import TableToJSON from util
+import Create, Exists, Remove, RepsLeft from timer
 export ChatTransit = {}
 
--- TODO: Relocate/clean these
-RelayPort = file.Read "cfc/cfc_relay_port.txt", "DATA"
-RelayPort = string.Replace RelayPort, "\r", ""
-RelayPort = string.Replace RelayPort, "\n", ""
+readClean = (fileName) ->
+    data = Read fileName, "DATA"
+    gsub data, "%s", ""
 
-RelayPassword = file.Read "cfc/cfc_relay_password.txt", "DATA"
-RelayPassword = string.Replace RelayPassword, "\r", ""
-RelayPassword = string.Replace RelayPassword, "\n", ""
-
-Realm = file.Read "cfc/realm.txt", "DATA"
-Realm = string.Replace Realm, "\r", ""
-Realm = string.Replace Realm, "\n", ""
+RelayPort = readClean "cfc/cfc_relay_port.txt"
+RelayPassword = readClean "cfc/cfc_relay_password.txt"
+Realm = readClean "cfc/realm.txt"
 
 ChatTransit.Logger = CFCLogger "CFC_ChatTransit"
 ChatTransit.TeamColorCache = {}
 ChatTransit.WebSocket = GWSockets.createWebSocket "ws://127.0.0.1:#{RelayPort}/relay"
 
 with ChatTransit.WebSocket
     Logger = ChatTransit.Logger
-    --\setHeader "Authorization", "Bearer #{RelayPassword}"
-    .onConnected = => Logger\info "Established websocket connection"
-    .onDisconnected = => Logger\warn "Lost websocket connection!"
+    .reconnectTimerName = "CFC_ChatTransit_WebsocketReconnect"
+
+    .onConnected = =>
+        Logger\info "Established websocket connection"
+        Remove .reconnectTimerName
+
+    .onDisconnected = =>
+        Logger\warn "Lost websocket connection!"
+
+        if Exists .reconnectTimerName
+            return Logger\warn "Will retry #{RepsLeft .reconnectTimerName} more times"
+
+        Create .reconnectTimerName, 2, 30, -> \open!
```

Comment:
This timer is only created on disconnect, does this mean if the websocket fails to connect on first attempt, it wont retry multiple times?
hmaybe changing the final `\open!` to like
```moonscript
\open!
Create .reconnectTimerName, 2, 30, -> \open!
```
would fix this? (if its actually a problem, not too sure how GWSockets works)

Reply:
Ah yes, I forgot to share the test data on this.

```lua
local TempSocket = GWSocket.createWebSocket( "ws://local:9999/blah" )

function TempSocket:onError( err )
    print("ERROR: " .. err)
end

function TempSocket:onDisconnected()
    print("DISCONNECTED")
end

TempSocket:open()
```

Resulted in:
```
ERROR: Connection failed: Connection refused
DISCONNECTED
```

So it appears to run the disconnect function even if it never actually connected to begin with 🤷

Reply:
Ah okay, coolio, looks good then!